test/v2/priceImpact/priceImpact.integration.test.ts

```diff
@@ -117,16 +136,62 @@ describe('price impact', () => {
         });
     });
 
+    describe('add liquidity nested - unbalanced', () => {
+        let input: AddLiquidityNestedInput;
+        beforeAll(() => {
+            input = {
+                chainId,
+                rpcUrl,
+                accountAddress: ZERO_ADDRESS,
+                amountsIn: [
+                    {
+                        address: DAI.address,
+                        rawAmount: parseUnits('100000', DAI.decimals),
+                        decimals: DAI.decimals,
+                    },
+                    {
+                        address: USDC.address,
+                        rawAmount: parseUnits('1000', USDC.decimals),
+                        decimals: USDC.decimals,
+                    },
+                    {
+                        address: USDT.address,
+                        rawAmount: parseUnits('10', USDT.decimals),
+                        decimals: USDT.decimals,
+                    },
+                    {
+                        address: WETH.address,
+                        rawAmount: parseUnits('0.1', WETH.decimals),
+                        decimals: WETH.decimals,
+                    },
+                ],
+            };
+        });
+        test('ABA close to Spot Price', async () => {
+            const priceImpactABA = await PriceImpact.addLiquidityNested(
+                input,
+                nestedPoolState,
+            );
+            const priceImpactSpot = PriceImpactAmount.fromDecimal(
+                '0.0472', // TODO: find a way to validate this result
```

Comment:
Something you've checked with Zen?

Reply:
Yeah, I'm actually using the old SDK results for comparison. I left this comment so I wouldn't forget to do it. Thanks for bringing my attention to it.
Ps: Zen was helping on validating the approach in a more generalized way, coming up with relevant scenarios for comparison and analyzing the results. Now that it's validated, it's easier to compare this scenario directly with results from the old SDK.